### ISSUE TITLE
Stops assigning hitbox's

### ISSUE THREAD

**Issue by Cart00nHead** [Toly65/UdonCombatSystem#16]
Hey
I recently updated everything and was testing a few things out.

I noticed that no matter how many hitbox's you have 2, 5 even the original 80 straight out of the box

After that many people have been through the world they stop getting assigned hitbox's? for them selves and everyone else 

do you know why this might be? 
perhaps they aren't retuning to the pool after PlayerLeft? 

Any help would be amazing 
Cheers CH

**Comment by Cart00nHead** [comment]
actually what i think is happening is its not removing ownership after the person leaves 🤔

**Comment by Toly65** [comment]
are you using the latest 0.6 version of the combat system?

**Comment by Cart00nHead** [comment]
Sure am, everything else works amazingly 

it seems once a new player id exceeds the number of hitboxes available it stops assigning them

**Comment by Toly65** [comment]
how many hitboxes do you have? and how many people are in the world at once?

**Comment by Cart00nHead** [comment]
I have tested it without touching anything with the 80 you have in a new project ect

and I have tested it with less. the number doesn't matter
 
say you have only 2 hit boxes in the array 
for 2 people
if 3 people join the same instance, then the 3rd wont get assigned anything for them selves or others that where previously in the world
not joining all at once but joining after someone as left
as long as 2 people have joined in it prior and it will mess up

I'm guessing because the new id from that person exceeds that of the number of hitboxes available
because with each person joining the number of ids gos up and up
untill BAM
no more ids no more hitboxs

**Comment by Toly65** [comment]
I'll have to check on how playerIDs work in VRchat
as far as I was aware old ones get reassigned so that they never really go above 80

**Comment by Toly65** [comment]
just checked, that's not the case, welp time to rewrite hitbox assignment again

**Comment by Cart00nHead** [comment]
😅 yeah nothing is ever straight forward with VRchat  

other than that, everything works great! you are doing a great job seriously!

**Comment by Toly65** [comment]
latest release should fix the issue

**Comment by Cart00nHead** [comment]
I just dropped it in a new unity project, Hit boxs seem to be reassigning now 👌

But I cant seem to do any dmg to the other players now? not sure what's happened 😅

**Comment by Toly65** [comment]
I would test and fix it however my entire development environment has failed so badly that I need to reinstall windows

**Comment by Cart00nHead** [comment]
I think I spoke to soon im so sorry

I manually changed the scripts in my original project ( Just the assigner and the Manager) to test and its working great now! not a single issue
Not sure why the demo scene in a new unity project was acting up 

And that's not fun at all I'm sorry to hear that, I went through a similar thing a few months back. 
Good luck! hope its an easy fix 😬

and Thanks again for the amazing work!
